cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -197,15 +197,21 @@ runGenesisCreateTestNetDataCmd Cmd.GenesisCreateTestNetDataCmdArgs
    , outputDir }
    = do
   liftIO $ createDirectoryIfMissing False outputDir
-  shelleyGenesis <-
+  shelleyGenesisInit <-
     case specShelley of
       Just shelleyPath ->
         newExceptT $ readAndDecodeShelleyGenesis shelleyPath
-      Nothing -> do
+      Nothing ->
         -- No template given: a default file is created
-        pure $ shelleyGenesisDefaults { sgNetworkMagic = unNetworkMagic (toNetworkMagic networkId) }
-
-  let -- {0 -> genesis-keys/genesis0/key.vkey, 1 -> genesis-keys/genesis1/key.vkey, ...}
+        pure shelleyGenesisDefaults
+  
+  -- Read NetworkId either from file or from the flag. Flag overrides template file.  
+  let actualNetworkId =
```

Comment:
In a follow up PR, you can refactor this with [maybe](https://hackage.haskell.org/package/base-4.19.0.0/docs/Prelude.html#v:maybe).